pkg/admission/handlers/pod.go

```diff
@@ -0,0 +1,67 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/kanopy-platform/hedgetrimmer/pkg/admission"
+	"github.com/kanopy-platform/hedgetrimmer/pkg/limitrange"
+	kadmission "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type PodHandler struct {
+	DefaultDecoderInjector
+	AllVersionSupporter
+	ptm admission.PodTemplateSpecMutator
+}
+
+func NewPodHandler(ptm admission.PodTemplateSpecMutator) *PodHandler {
+	return &PodHandler{ptm: ptm}
+}
+
+func (p *PodHandler) Kind() string {
+	return "Pod"
+}
+
+func (p *PodHandler) Handle(ctx context.Context, req kadmission.Request) kadmission.Response {
```

Comment:
Should this handler have an additional conditional to skip pods that have an OwnerReference ?
e.g
```
 ownerReferences:
  - apiVersion: batch/v1
    blockOwnerDeletion: true
    controller: true
    kind: Job
    name: pi
    uid: a6bde40a-5f19-4a9c-8c0c-4db133827279
```

This is a sample reference from a pod spawned from a job.

Reply:
This should be idempotent for any resource that hedgetrimmer has already touched since they will have requests and limits set already. We could add another short circuit but we would need to define all the logic. Pods can own other Pods for instance. Also we are handling the default replication controllers but if something else creates a pod directly we would need a list of excluded owner kinds.